src/Compilers/Core/Portable/Collections/ImmutableArrayDictionary`2.cs

```diff
@@ -0,0 +1,296 @@
+            dictionary.Add(key, value);
+            return new ImmutableArrayDictionary<TKey, TValue>(dictionary, _valueComparer);
+        }
+
+        public ImmutableArrayDictionary<TKey, TValue> AddRange(IEnumerable<KeyValuePair<TKey, TValue>> pairs)
+        {
+            var dictionary = new Dictionary<TKey, TValue>(_dictionary, _dictionary.Comparer);
+            foreach (var (key, value) in pairs)
+            {
+                dictionary.Add(key, value);
+            }
+
+            return new ImmutableArrayDictionary<TKey, TValue>(dictionary, _valueComparer);
+        }
+
+        public ImmutableArrayDictionary<TKey, TValue> Clear()
+        {
+            if (IsEmpty)
+            {
+                return this;
+            }
+
+            return Empty.WithComparers(KeyComparer, ValueComparer);
+        }
+
+        public bool Contains(KeyValuePair<TKey, TValue> pair)
+        {
+            return TryGetValue(pair.Key, out TValue value)
+                && ValueComparer.Equals(value, pair.Value);
+        }
+
+        public bool ContainsKey(TKey key)
+            => _dictionary.ContainsKey(key);
+
+        public bool ContainsValue(TValue value)
+        {
+            if (ValueComparer == EqualityComparer<TValue>.Default)
+            {
+                return _dictionary.ContainsValue(value);
+            }
+            else
+            {
+                foreach (var pair in this)
+                {
+                    if (ValueComparer.Equals(pair.Value, value))
+                        return true;
+                }
+
+                return false;
+            }
+        }
+
+        public Enumerator GetEnumerator()
+            => new Enumerator(_dictionary, Enumerator.ReturnType.KeyValuePair);
+
+        public ImmutableArrayDictionary<TKey, TValue> Remove(TKey key)
+        {
+            if (!_dictionary.ContainsKey(key))
+                return this;
+
+            var dictionary = new Dictionary<TKey, TValue>(_dictionary, _dictionary.Comparer);
+            dictionary.Remove(key);
+            return new ImmutableArrayDictionary<TKey, TValue>(dictionary, _valueComparer);
+        }
+
+        public ImmutableArrayDictionary<TKey, TValue> RemoveRange(IEnumerable<TKey> keys)
+        {
+            if (keys is null)
+                throw new ArgumentNullException(nameof(keys));
+
+            var result = ToBuilder();
+            result.RemoveRange(keys);
+            return result.ToImmutable();
+        }
+
+        public ImmutableArrayDictionary<TKey, TValue> SetItem(TKey key, TValue value)
+        {
+            var dictionary = new Dictionary<TKey, TValue>(_dictionary, _dictionary.Comparer);
+            dictionary[key] = value;
+            return new ImmutableArrayDictionary<TKey, TValue>(dictionary, _valueComparer);
+        }
+
+        public ImmutableArrayDictionary<TKey, TValue> SetItems(IEnumerable<KeyValuePair<TKey, TValue>> items)
```

Comment:
>SetItems [](start = 54, length = 8)

I'd suggest we leave all mutation APIs required by interfaces as unimplemented (ie. throwing in `SetItem`, `SetItems`, `Clear`, `Add`, `AddRange`, `Remove`, `RemoveRange`, and possibly even `WithComparers`) and remove the `ToBuilder` API.

Reply:
➡️ This pull request intentionally exposes the full API surface area.